src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -300,14 +298,14 @@ static String guessModByFileName(String fileName) {
       return CompletableFuture.allOf(featuredModFuture, mapBeanFuture).thenApply(ignoredVoid  -> {
         Optional<MapBean> mapBean = mapBeanFuture.join();
         if (!mapBean.isPresent()) {
-          throw new CompletionException(new FileNotFoundException());
+          logger.warn("Could not find map for replay file '{}'", replayFile);
         }
-        return new Replay(replayInfo, replayFile, featuredModFuture.join(), mapBean.get());
+        return new Replay(replayInfo, replayFile, featuredModFuture.join(), mapBean.orElse(null));
       });
     } catch (Exception e) {
       logger.warn("Could not read replay file '{}'", replayFile, e);
       moveCorruptedReplayFile(replayFile);
-      return CompletableFuture.failedFuture(e);
+      return CompletableFuture.completedFuture(null);
```

Comment:
Don't throw others wise replay vault stays empty, return null which is gonna be filtered out